index.js

```diff
@@ -47,7 +64,7 @@ function co(gen) {
   // we wrap everything in a promise to avoid promise chaining,
   // which leads to memory leak errors.
   // see https://github.com/tj/co/issues/180
-  return new Promise(function(resolve, reject) {
+  return new coPromise(function(resolve, reject) {
```

Comment:
This is a very good question; my reasoning for keeping this is twofold:

1. The two promise libraries I've looked at (Q and bluebird) both support this method, though Q requires you to use Q.Promise (and yes, I've tested it and it works)

2. It is trivially easy to make a wrapper around any promise library which will support this interface; some libraries may require that anyway in order to support all of the specific function names which co uses.  I figure this is fair, though, since co uses the ES6 standard so any promise library wishing to use it will just need to either support that interface or have a wrapper which allows it to.